web/base_viewer.js

```diff
@@ -262,6 +265,18 @@ class BaseViewer {
     this.maxCanvasPixels = options.maxCanvasPixels;
     this.l10n = options.l10n || NullL10n;
     this.#enablePermissions = options.enablePermissions || false;
+    this.pageColors = options.pageColors || null;
+
+    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
+      if (
+        options.pageColors &&
+        options.pageColors.foreground === "CanvasText" &&
+        options.pageColors.background === "Canvas" &&
+        !FeatureTest.isCSSColorCanvasTextSupported
+      ) {
```

Comment:
You might want to add a warning in this case, e.g.
```
console.warn("Ignoring `pageColors`-option, since the browser doesn't support the values used.");
```